File: server/src/Services/EventService/EventService.API/Entities/Event.cs
namespace EventService.API.Entities;

public class Event
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string? Image { get; set; }
    public EventStatus Status { get; set; } = EventStatus.Pending;
    public string? Comment { get; set; }
    public DateTimeOffset StartTime { get; set; }
    public DateTimeOffset EndTime { get; set; }
    public Guid BrandId { get; set; }

    public List<VoucherType> VoucherTypes { get; set; } = new();
    public int TotalVouchers => VoucherTypes.Sum(voucherType => voucherType.Total);
    public int TotalPublishedVouchers => VoucherTypes.Sum(voucherType => voucherType.Total - voucherType.Remaining);

    public List<EventGame> Games { get; set; } = new();
    public Item? Item { get; set; }
}

File: client/mobile/vousphere/lib/data/models/Event.dart
import 'package:vousphere/data/models/Brand.dart';

class Event {
  final String id;
  final String name;
  final String description;
  final String image;
  final String status;
  final DateTime startTime;
  final DateTime endTime;
  final Brand brand;
  final List<Map<String, dynamic>> voucherTypes;
  final int totalVouchers;
  final int totalPublishedVouchers;
  final List<Map<String, dynamic>> games;

  Event({
    required this.id,
    required this.name,
    required this.description,
    required this.image,
    required this.status,
    required this.startTime,
    required this.endTime,
    required this.brand,
    required this.voucherTypes,
    required this.totalVouchers,
    required this.totalPublishedVouchers,
    required this.games,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      image: json['image'] ?? '',
      status: json['status'] ?? '',
      startTime:
          DateTime.parse(json['startTime'] ?? DateTime.now().toIso8601String())
              .toLocal(),
      endTime:
          DateTime.parse(json['endTime'] ?? DateTime.now().toIso8601String())
              .toLocal(),
      brand: Brand.fromJson(json['brand'] ?? {}),
      voucherTypes: List<Map<String, dynamic>>.from(
          json['voucherTypes']?.map((item) => item as Map<String, dynamic>) ??
              []),
      totalVouchers: json['totalVouchers'] ?? 0,
      totalPublishedVouchers: json['totalPublishedVouchers'] ?? 0,
      games: List<Map<String, dynamic>>.from(
          json['games']?.map((item) => item as Map<String, dynamic>) ?? []),
    );
  }
}
